hnn_core/optimization/optimizer.py

```diff
@@ -0,0 +1,382 @@
+"""Parameter optimization."""
+
+# Authors: Carolina Fernandez <cxf418@miami.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Ryan Thorpe <ryan_thorpe@brown.edu>
+#          Mainak Jas <mjas@mgh.harvard.edu>
+
+import numpy as np
+
+from .metrics import _rmse_evoked
+
+from skopt import gp_minimize
+from scipy.optimize import fmin_cobyla
+
+
+class Optimizer:
+    def __init__(self, net, constraints, set_params, solver, obj_fun,
+                 tstop, scale_factor=1., smooth_window_len=None):
```

Comment:
you have a lot of required arguments for the user ... can we set some sensible defaults? Suggestion below:

```suggestion
    def __init__(self, net, tstop, constraints, set_params, solver='cobyla', obj_fun='evoked_rmse',
                        scale_factor=1., smooth_window_len=None):
```